libraries/botbuilder-dialogs-adaptive/src/adaptiveDialogBot.ts

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import memoize from 'lodash/memoize';
+import { AdaptiveDialog } from './adaptiveDialog';
+import { BotFrameworkAuthentication, BotFrameworkClient, BotFrameworkClientKey } from 'botframework-connector';
+import { Dialog, MemoryScope, PathResolver, runDialog } from 'botbuilder-dialogs';
+import { LanguagePolicy } from './languagePolicy';
+import { ResourceExplorer } from 'botbuilder-dialogs-declarative';
+import { languageGeneratorKey, languageGeneratorManagerKey, languagePolicyKey } from './languageGeneratorExtensions';
+import { resourceExplorerKey } from './resourceExtensions';
+import { skillConversationIdFactoryKey } from './skillExtensions';
+
+import {
+    ActivityHandler,
+    ActivityTypes,
+    BotCallbackHandlerKey,
+    BotTelemetryClient,
+    BotTelemetryClientKey,
+    ConversationState,
+    SkillConversationIdFactoryBase,
+    TurnContext,
+    UserState,
+} from 'botbuilder';
+
+import {
+    LanguageGeneratorManager,
+    ResourceMultiLanguageGenerator,
+    TemplateEngineLanguageGenerator,
+} from './generators';
+
+export class AdaptiveDialogBot extends ActivityHandler {
+    private static readonly languageGeneratorManagers = new Map<ResourceExplorer, LanguageGeneratorManager>();
+
+    private readonly lazyRootDialog: () => Dialog;
+
+    constructor(
+        private readonly adaptiveDialogId: string,
+        private readonly languageGeneratorId: string,
+        private readonly resourceExplorer: ResourceExplorer,
+        private readonly conversationState: ConversationState,
+        private readonly userState: UserState,
+        private readonly skillConversationIdFactoryBase: SkillConversationIdFactoryBase,
+        private readonly languagePolicy: LanguagePolicy,
+        private readonly botFrameworkAuthentication: BotFrameworkAuthentication,
+        private readonly telemetryClient: BotTelemetryClient,
+        private readonly memoryScopes: MemoryScope[] = [],
+        private readonly pathResolvers: PathResolver[] = [],
+        private readonly dialogs: Dialog[] = []
+    ) {
+        super();
+
+        this.lazyRootDialog = memoize(() => this.createDialog());
+
+        this.onTurn(async (context, next) => {
```

Comment:
Is this intentional? This implementation uses the event listener pattern which imo, doesn't accurately represent all of the Activity processing scenarios. The implementation could instead override `ActivityHandler.onTurnActivity(context: TurnContext): Promise<void>` since this is the only `on*` method being overridden and I think, used.

https://github.com/microsoft/botbuilder-js/blob/29f86bb7f62b11253624e1692b4612148489936e/libraries/botbuilder-core/src/activityHandlerBase.ts#L51

Additionally, the .NET implementation (`AdaptiveDialogBot : IBot`) just implements its own OnTurnAsync method, and doesn't have the rest of the ActivityHandler "luggage" brought along with it.

I might be wrong on this, but the `IBot` and `BotCallbackHandler` types in .NET are more prevalent and passed around throughout the SDK than the `ActivityHandler`, which is one reason why in .NET `AdaptiveDialogBot` implements `IBot` instead of the `ActivityHandler`.

Reply:
Note: Not "Outdated". I moved this all to an `onTurnActivity()` override. Still need to address `IBot`.

Reply:
Working on `IBot` as we speak.